inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -252,60 +374,65 @@ std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::compose_traject
-
-  ROS_DEBUG("Appended to final");
   
-  ROS_DEBUG("Processed all points in computed fit");
+  //ROS_DEBUG("Processed all points in computed fit");
 
   if (all_sampling_points.size() == 0)
   {
     ROS_WARN_STREAM("No trajectory points could be generated");
     return {};
   }
 
-  log::printDoublesPerLineWithPrefix("final_actual_speeds[i]: ", final_actual_speeds);
-
-  log::printDoublesPerLineWithPrefix("final_yaw_values[i]: ", final_yaw_values);
+  //log::printDoublesPerLineWithPrefix("final_actual_speeds[i]: ", final_actual_speeds);
 
-  // Find Lookahead Distance based on Velocity
-  double lookahead_distance = get_adaptive_lookahead(state.longitudinal_vel);
-
-  ROS_DEBUG_STREAM("Lookahead distance at current speed: " << lookahead_distance);
+  //log::printDoublesPerLineWithPrefix("final_yaw_values[i]: ", final_yaw_values);
 
   // Apply lookahead speeds
-  final_actual_speeds = get_lookahead_speed(all_sampling_points, final_actual_speeds, lookahead_distance);
-  
+  // final_actual_speeds = trajectory_utils::shift_by_lookahead(final_actual_speeds, config_.lookahead_count);
+
   // Add current vehicle point to front of the trajectory
+
+  nearest_pt_index = get_nearest_point_index(all_sampling_points, state);
+  ROS_DEBUG_STREAM("Curvature right now: " << better_curvature[nearest_pt_index] << ", at x: " << state.X_pos_global << ", y: " << state.Y_pos_global);
+
+  std::vector<lanelet::BasicPoint2d> future_basic_points(all_sampling_points.begin() + nearest_pt_index + 1,
+                                            all_sampling_points.end());  // Points in front of current vehicle position
+
+  std::vector<double> future_speeds(final_actual_speeds.begin() + nearest_pt_index + 1,
+                                            final_actual_speeds.end());  // Points in front of current vehicle position
+  std::vector<double> future_yaw(final_yaw_values.begin() + nearest_pt_index + 1,
+                                            final_yaw_values.end());  // Points in front of current vehicle position
+  final_actual_speeds = future_speeds;
+  all_sampling_points = future_basic_points;
+  final_yaw_values = future_yaw;
+
   lanelet::BasicPoint2d cur_veh_point(state.X_pos_global, state.Y_pos_global);
 
   all_sampling_points.insert(all_sampling_points.begin(),
                              cur_veh_point);  // Add current vehicle position to front of sample points
 
-  final_actual_speeds.insert(final_actual_speeds.begin(), std::max(state.longitudinal_vel, config_.minimum_speed));
+  final_actual_speeds.insert(final_actual_speeds.begin(), state.longitudinal_vel);
 
   final_yaw_values.insert(final_yaw_values.begin(), state.orientation);
 
-  log::printDoublesPerLineWithPrefix("pre_smoot[i]: ", final_actual_speeds);
+  //log::printDoublesPerLineWithPrefix("pre_smoot[i]: ", final_actual_speeds);
   
   // Compute points to local downtracks
   std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(all_sampling_points);
```

Comment:
I think there might be a bug here. I believe this should be called on the future_points with the curv_veh_point appended to the front. This looks like it still includes the back points, but I might have missed something.

Reply:
yeah I confirmed that it is applied to the [curr_point + future_points]. looks like there is assignment below just before this. I'll make it clearer now that we don't have sub_curves.
`  final_actual_speeds = future_speeds;
  all_sampling_points = future_basic_points;
  final_yaw_values = future_yaw;`